logbook-api/src/test/java/org/zalando/logbook/ContentTypeTest.java

```diff
@@ -15,11 +15,15 @@ void unitTest() {
         assertThat(ContentType.parseMimeType("")).isNull();
 
         assertThat(ContentType.parseCharset(null)).isNull();
-        assertThat(ContentType.parseCharset("application/json; charset=")).isNull();
-        assertThat(ContentType.parseCharset("application/json; charset=unknown-charset")).isNull();
         assertThat(ContentType.parseCharset("charset=;")).isNull();
         assertThat(ContentType.parseCharset("charset=foo\"")).isNull();
         assertThat(ContentType.parseCharset("charset=\"bar")).isNull();
         assertThat(ContentType.parseCharset("application/json; charset=\"us-ascii\"")).isEqualTo(StandardCharsets.US_ASCII);
     }
+
+    @Test
+    void fallbackToUTF8WhenJson() {
+        assertThat(ContentType.parseCharset("application/json; charset=")).isEqualTo(StandardCharsets.UTF_8);
+        assertThat(ContentType.parseCharset("application/json; charset=unknown-charset")).isEqualTo(StandardCharsets.UTF_8);
```

Comment:
- Is the same logic applied if the content type is, e.g., `application/problem+json`? If so, may I suggest adding a test for it as well?
- Also, can we add a test to make sure content-type matching is **not** case sensitive? (both for header name `Content-Type:`/`content-type:` and its value `Application/JSON` / `application/json`)

Reply:
That's a very good catch! Indeed, there was a bug in recognizing Content-Type headers and recognizing JSON Content-Type headers. But according to the RFC 9110 they should be case-insensitive. I added more test cases and fixed it in https://github.com/zalando/logbook/pull/1521/commits/0d7cda1162867131d40003f952b9b5bd796cbef3

Reply:
Thanks!
One more suggestion: Could something like this simplify the nested-if logic?

```java
public static String getSubstringBeforeSemicolon(String s) {
    final int index = s.indexOf(';');  // not sure if this should be: s.indexOf(';', 12)
    if (index != -1) return s.substring(0, index);
    return s;
}
```